File: LAB QUIZZES/CSE221_Quiz02/_A2A/_A2A_Testing.java
import java.util.*;
import java.io.*;

// Compile and run: javac _A2A_Testing.java && java _A2A_Testing
class _A2A_Solution {
    public static class Pair<K, V> {
        public K key;
        public V value;

        public Pair(K key, V value) {
            this.key = key;
            this.value = value;
        }

        public String toString() {
            return "(" + key + ", " + value + ")";
        }
    }

    public static Pair<Integer, Integer> solve(Integer N, Integer S, ArrayList<Integer> A) {
        Integer I = 0, J = 0;
        // Place your code here
        return new Pair<>(I, J);
    }
}

public class _A2A_Testing {
    Random rng = new Random();
    public static Integer N = 0, S = 0;
    public static ArrayList<Integer> A = new ArrayList<>();
    public static _A2A_Solution.Pair<Integer, Integer> res;

    public static void main(String args[]) throws Exception {
        // // stdin
        // // 4 10
        // // 1 3 5 7
        BufferedReader fin = new BufferedReader(new InputStreamReader(System.in));
        String[] temp2 = fin.readLine().split(" ");
        N = Integer.parseInt(temp2[0]);
        S = Integer.parseInt(temp2[1]);
        for (String temp : fin.readLine().split(" "))
            A.add(Integer.parseInt(temp));

        // // hardcoded input
        // N = 4
        // S = 10
        // A = new ArrayList<Integer>({1, 3, 5, 7});

        long start = System.nanoTime();
        res = _A2A_Solution.solve(N, S, new ArrayList(A));
        long finish = System.nanoTime();
        long elapsed = finish - start;

        // stdout
        System.out.println(res.toString());

        System.out.printf("Execution time for the targeted one test is %.9fs\n", elapsed * 1e-9);
        System.exit(0);

        // // file input
        // BufferedReader fin = new BufferedReader(new
        // FileReader("TimeLimitExceeded.txt"));
        // BufferedReader fin = new BufferedReader(new FileReader("CustomTest.txt"));
        // BufferedReader fin = new BufferedReader(new FileReader("WrongAnswer.txt"));
        // for (int i = 1; i <= 2; ++i)
        // fin.readLine();
        // N = Integer.parseInt(fin.readLine().split(" ")[2]);
        // S = Integer.parseInt(fin.readLine().split(" ")[2]);
        // String[] temp = fin.readLine().split(" ");
        // for(int i = 0; i < N; i++){
        // if(i == 0)
        // A.add(Integer.parseInt(temp[2].substring(1, temp[2].length() - 1)));
        // else
        // A.add(Integer.parseInt(temp[i + 2].substring(0, temp[i + 2].length() - 1)));
        // }
        // fin.close();

        // // file output
        // BufferedWriter fout = new BufferedWriter(new
        // FileWriter("OutputOfYourCode.txt"));
        // fout.write(res.toString());
        // fout.close();
    }
}

File: LAB QUIZZES/CSE221_Quiz02/_A2A/_A2A_Testing.py
import os, sys, time, random, threading, subprocess
N = 0
S = 0
A = []
# Compile and run: pypy _A2A_Testing.py
def solve(N, S, A):
    flag = False
    i, j = 0, len(A) - 1
    while i <= j:
        sum = A[i] + A[j]
        if sum == S:
            flag = True
            break
        if sum > S:
            j -= 1
        else:
            i += 1
    if flag == False:
        return (-1, -1)
    return (i + 1, j + 1)


# stdin
# 4 10
# 1 3 5 7
N, S = list(map(int, "9 27".split()))
A = [35, 219, 321, 350, 501, 527, 757, 838, 845]


# # hardcoded input
# N, S = 4, 10
# A = [1, 3, 5, 7]

start = time.time()
res = solve(N, S, A.copy())
finish = time.time()
elapsed = finish - start

# # stdout
print(res)

print("Execution time for the targeted one test is " + f"{elapsed:.9f}" + "s", flush=True)
os._exit(0)
# # file input
# with open("TimeLimitExceeded.txt") as fin:
# with open("CustomTest.txt") as fin:
# with open("WrongAnswer.txt") as fin:
#     lines = fin.readlines()
#     N = int(lines[2].split(" ")[2])
#     S = int(lines[3].split(" ")[2])
#     A = list(map(int, lines[4][5:-2].split(", ")))

# # file output
# with open("OutputOfYourCode.txt", "w") as fout:
#     fout.write(str(res))
